README
Eekboek Boekhoudpakket
======================

EekBoek is een boekhoudprogramma bedoeld voor het midden- en
kleinbedrijf.

EekBoek is niet zomaar een conventioneel boekhoudprogramma. EekBoek
is ontstaan uit de gedachte die iedere wat meer technisch onderlegde
bedrijfsvoerder wel eens heeft gehad: "Boekhouden, da's toch
gewoon wat getalletjes organiseren? Dat moet toch gemakkelijk kunnen
met wat slimme scriptjes."

Velen hebben pogingen daartoe gedaan, maar de meesten faalden omdat
het toch iets complexer bleek te zijn. De goedbedoelde 'slimme
scriptjes' werden weer afgedankt, en men nam noodgedwongen een
commercieel, meestal op Microsoft Windows gebaseerd boekhoudpakket.

EekBoek is wel geslaagd in deze opzet. De 'slimme scriptjes' zijn
weliswaar uitgegroeid tot een stevige applicatie, maar het geheel
vertoont nog steeds alle kenmerken die zo belangrijk waren: eenvoudig
in gebruik, command line gedreven, en een programmeerbare API.

EekBoek slaat alle data op in een relationele database, wat het geheel
robuust en betrouwbaar maakt. De structuur van de administratie en de
boekingsopdrachten kunnen worden onderhouden in twee eenvoudige
tekstbestanden. Voor de andersdenkenden is een grafische user
interface in ontwikkeling die op dezelfde database werkt, Een
export-faciliteit kan van alles wat met de GUI is uitgevoerd weer
tekstbestandjes maken. Op deze wijze blijft het geheel overzichtelijk.

Na een uitgebreide test-periode is EekBoek nu voldoende capabel om aan
het brede publiek te worden gepresenteerd.

Voor meer informatie: http://www.squirrel.nl/eekboek .

Voor het uitwisselen van ervaringen, vragen om ondersteuning en zo:
gebruik de mailing list http://lists.sourceforge.net/lists/listinfo/eekboek-users.

----------------------------------------------------------------------------
Johan Vromans                                           jvromans@squirrel.nl
Squirrel Consultancy                                  Exloo, the Netherlands
http://www.squirrel.nl                              http://johan.vromans.org
PGP Key 2048/4783B14D                   http://johan.vromans.org/pgpkey.html
----------------------- "Arms are made for hugging" ------------------------


CHANGES
2.00.04

	* Fix bug 3499522.

	  Een IV boeking waarbij een BTW tariefgroep Privé of Anders
	  was betrokken, werd abusievelijk op de grootboekrekening
	  voor BTW Laag geboekt. In geval van een verkoopboeking kwam
	  het BTW bedrag niet in de juiste rubriek van de BTW aangifte
	  terecht.

	  (Voor inkoopboekingen was de impact op de BTW aangifte
	  nihil.)

	* Fix fout bij opvragen hulp voor BTW aangifte.

	* Fix typo in enkele boodschappen (Issue #3477452).

2.00.03	2011-03-22

	* (schema) Fix probleem dat het gebruik van een
	  ongedefinieerde BTW tariefgroep onopgemerkt bleef wanneer er
	  wel een groep was gedefinieerd met een afwijkende
	  exclusiviteit.

2.00.02	2010-05-06

	* (ebwxshell, linux) Workaround voor het HTML rendering
          probleem met wxPerl versies 0.95 en 0.95.
	  Dit probleem treft met name Ubuntu 10.4 LTS.

	* (ebshell) Meer flexibele invoer.
	  Accepteer newlines en returns als regeleinde.
	  Negeer trailing whitespace. Dit elimineert het probleem van
	  spaties achter de laatste backslash.

	* (eekboek-mode.el) Kleine, niet strict noodzakelijke
	  verbetering. 

	* (ebshell, errexit) Invoerverwerking afbreken indien er een
	  foutboodschap is gegeven.

	* INSTALL: Refereer naar de wiki.

2.00.01.01

	* Packaging for Windows: (build_common.pl) use PL_files to
          avoid .pl files to be processed.

2.00.01	2010-03-28 20:42

	* Fix probleem met Term::ReadLine::Perl. Gebruik in plaats
          daarvan Term::ReadLine::Gnu.
	  Meest relevant voor Windows en MacOSX.

	* Elimineer File::HomeDir.
	  Relevant voor MacOSX, omdat de benodigde Mac::Files daar
	  problematisch is en enkel in 32bit beschikbaar.

	* Nieuwe packaging. Default is cli, gui is optioneel, contrib
          verhuist naar de wiki.

2.00.00	2010-03-23 20:59

	* Release 2.00.00

2.00.00.RC03	2010-03-22 14:04

	* Fix probleem dat een overbodige dialoog vraagt om een
          eekboek.conf.

2.00.00.RC02	2010-03-17 13:40

	* Tweede release candidate 2.00.

	* Diverse fixes voor Windows.

	* Fix probleem met niet eenduidige interpretatie van getallen.
	  In expressies als "5*0.005" werd 0.005 als 5 gezien i.p.v.
	  0,005. 

	* Kleine verbeteringen in de wizards. 

2.00.RC01	2010-03-12 23:13

	* Eerste release candidate 2.00.

	* (iniwiz) Fix dubbele foutmelding.
	
	* Kleine, maar belangrijke aanpassingen t.b.v. Microsoft
	  Windows.

	* Onderhoud vertalingen toegevoegd aan de kit.

1.05.20	2010-02-19 23:26

	* Documentatie aanpassingen.

	* (iniwiz) Fix probleem met crash na db selectie.

	* Fix probleem met \ op laatste regel.

	(wbwxshell) Fix probleem met de result rapporten.

1.05.19	2010-01-27 10:25

	* (ebshell) Fix unicode probleem in help strings.

	* (ebshell) Fix unicode probleem in wizard waardoor
          administratie namen/codes geen non-ascii tekens konden
          bevatten.

1.05.18	2010-01-26 10:56

	* (ebwxshell) Fix unicode issues.

	* (ebwxshell) Filter duplicaten uit de history en beperk het
          aantal teruggeschreven regels tot een instelbaar maximum
          (default 200).

	* Upgrade Wx::Perl::ProcessStream to 0.24.

1.05.17	2010-01-22 13:10

	* Nieuw command: schema. Hiermee kunnen interactief
          (vooralsnog alleen) grootboekrekeningen worden aangemaakt.

	  Onderhoud van het schema. Deze opdracht kent sub-opdrachten:

	    schema <opdracht> [ <opties> ] <argumenten>

	  Aanmaken grootboekrekening

	    schema gbk <rekening> [ <type> <omschrijving> <verdichting> ]

	       <rekening>      de gewenste grootboekrekening
	       <type>          D/C voor Debet / Credit
			       K/O/N voor Kosten / Omzet / Neutraal
			       Eventueel gevolgd door ! als deze 
			       balansrekening vast staat aan één kant
	       <omschrijving>  De omschrijving van deze grootboekrekening
	       <verdichting>   De verdichting waaronder deze rekening valt

	       Wanneer enkel een nummer wordt opgegeven dan worden de gegevens
	       van de betreffende grootboekrekening getoond.

	* Fix probleem met completion van rekeningnummers.

	* Upgrade Module::Build to 0.32 to have decent prereq support.

1.05.16 2010-01-17 00:14

	* (ebshell) Non-zero exit status indien fouten zijn
          opgetreden. Dit kan consequenties hebben als ebshell wordt
          toegepast in makefiles.

	* (ebshell) Verwijderen opties --createsampleconfig en
          --createsampledb (vervangen door de wizard).

	* (ebshell, wizard) Gebruik eigen code (recursief) om --init
	  uit te voeren.

	* (ebshell, shell) Grote opruiming.

	* (migration) Fix: DB automigratie van SQLite zat PostgreSQL
          in de weg.

1.05.15 2010-01-16 16:32

	* Small changes for the packaging.

1.05.14 2010-01-13 11:12

	* (ebwxshell) Icon buttons i.p.v. [Edit] en [Send].
	  Conform to Gnome Human Interface Guide.

	* (ebwxshell, editdialoog) Resizable, onthoudt size/pos.

	* (ebwxshell) Meer rapporten in het menu.

	* (ebwxshell) EditDialog meeleveren :)

	* Eerste herziening van de documentatie.

	* Fix test probleem op een virgo install.

	* Fix probleem met het verwerken van het schema (privé werd
          niet herkend).

	* Kleine aanpassingen om het opstartgedrag van ebshell en
          ebwxshell consistent te maken.

1.05.13 2010-01-06 21:21
1.05.12 2010-01-06 20:09

	=== EXPERIMENTAL ===

	* De configuratie file mag nu ook "eekboek.conf" heten.

	  Sterker nog, "eekboek.conf" wordt geprefereerd boven
	  ".eekboek.conf". Dus wanneer beide er zijn wordt alleen
	  "eekboek.conf" gebruikt. Wanneer nieuwe files worden
	  aangemaakt wordt alleen "eekboek.conf" aangemaakt.

	  Wanneer er alleen ".eekboek.conf" bestanden zijn moet het
	  gedrag ongewijzigd zijn gebleven.

1.05.11	2010-01-06 13:56

	* (ebwxshell, wizard) Verplicht unieke administratie naam en code.

	* (ebshell) Wizard om administraties aan te maken.

	* (ebwxshell) Mogelijkheid om de input regel uitgebreid te
	  editen, inclusief multi-line support.

1.05.10	2009-12-23 22:32

	* (ebwxshell) Persistent window size/positions.


Build.PL
# Build.PL -- Buildfile for EekBoek	-*-perl-*-
# Author          : Johan Vromans
# Created On      : Thu Sep  1 17:28:26 2005
# Last Modified By: Johan Vromans
# Last Modified On: Tue Sep 23 19:51:22 2008
# Update Count    : 115
# Status          : Experimental

# Ease the life of the CPAN testers.
exit 0 if $ENV{AUTOMATED_TESTING} && $] < 5.008003;

use 5.008003;
use strict;
use lib 'inc';
use Module::Build 0.27_09;
use File::Find;

our $data = {};
require "inc/build_common.pl";

# Warn on non-standard install locations.
checkbin(<<EOD);
Please consult section "How Installation Paths are Determined"
of the Module::Build documentation section if you want to change
this location.
EOD

# checkexec("psql");

my $class = Module::Build->subclass(code => <<'EOF');
sub ACTION_test {
    my $self = shift;
    if ( $self->args("skipdbtests") ) {
	warn("WARNING: Skipping database tests\n");
	$ENV{EB_SKIPDBTESTS} = 1;
    }
    $self->SUPER::ACTION_test;
}
EOF

my $build = $class->new
  (
   module_name	       => $data->{distname},
   dist_name	       => $data->{distname},
   dist_author	       => $data->{author},
   dist_abstract       => $data->{abstract},
   license	       => $data->{license},
   requires	       => $data->{prereq_pm},
   build_requires      => $data->{buildreq_pm},
   recommends	       => $data->{recomm_pm},
   script_files	       => $data->{script_files},
   pm_files	       => filelist("lib"),
   get_options	       => { skipdbtests => { } },
   add_to_cleanup      =>
     [ 'pod2htm*',
       map { +"t/ivp/*.$_" } qw(sql log txt html csv)
     ],
   meta_merge	       =>
     { resources =>
	{ homepage    => 'http://www.eekboek.nl',
	  MailingList => 'http://lists.sourceforge.net/lists/listinfo/eekboek',
	}
     },
   meta_add            => {
     no_index          => { directory => [ qw(inc lib) ] },
     provides	       => {
       'EekBoek' =>
	  { file => "lib/EekBoek.pm",
	    version => "1.04.00_01",
	  },
     },
   },
  );

# Rerun to get the version right for provides => EeoBoek.
$build = $class->new
  (
   module_name	       => $data->{distname},
   dist_name	       => $data->{distname},
   dist_author	       => $data->{author},
   dist_abstract       => $data->{abstract},
   license	       => $data->{license},
   requires	       => $data->{prereq_pm},
   build_requires      => $data->{buildreq_pm},
   recommends	       => $data->{recomm_pm},
   script_files	       => $data->{script_files},
   pm_files	       => filelist("lib"),
   get_options	       => { skipdbtests => { } },
   add_to_cleanup      =>
     [ 'pod2htm*',
       map { +"t/ivp/*.$_" } qw(sql log txt html csv)
     ],
   meta_merge	       =>
     { resources =>
	{ homepage    => 'http://www.eekboek.nl',
	  MailingList => 'http://lists.sourceforge.net/lists/listinfo/eekboek',
	}
     },
   meta_add            => {
     no_index          => { directory => [ qw(inc lib) ] },
     provides	       => {
       'EekBoek' =>
	  { file => "lib/EekBoek.pm",
	    version => $build->dist_version,
	  },
     },
   },
  );

$build->create_build_script;

if ( $build->os_type eq "Unix" ) {
    # Write .spec file, Debian control, ...
    ProcessTemplates($build->dist_name, $build->dist_version);
}

# Debian needs this.
1;
